Backpack/Nudger/Classes/BPKNudger.m

```diff
@@ -0,0 +1,297 @@
+
+- (void)decrementValue {
+    self.value -= self.stepValue;
+}
+
+- (BOOL)canDecrement {
+    return (self.value - self.stepValue) >= self.minimumValue;
+}
+
+- (void)incrementValue {
+    self.value += self.stepValue;
+}
+
+- (BOOL)canIncrement {
+    return (self.value + self.stepValue) <= self.maximumValue;
+}
+
+- (void)setValue:(double)value {
+    if (value != _value) {
+        if (value < self.minimumValue || value > self.maximumValue) {
+            return;
+        }
+
+        _value = value;
+        [self updateDisplay];
+    }
+}
+
+- (void)setMinimumValue:(double)minimumValue {
+    if (minimumValue > self.maximumValue) {
+        NSString *reason = [NSString stringWithFormat:@"Cannot set minimumValue %f when maximumValue is %f",
+                                    minimumValue, self.maximumValue];
+        @throw [NSException exceptionWithName:NSInvalidArgumentException
+                                       reason:reason
+                                     userInfo:nil];
+    }
+
+    _minimumValue = minimumValue;
+
+    [self updateDisplay];
+}
+
+- (void)setMaximumValue:(double)maximumValue {
+    if (maximumValue < self.minimumValue) {
+        NSString *reason = [NSString stringWithFormat:@"Cannot set maximumValue %f when minimumValue is %f",
+                                    maximumValue, self.minimumValue];
+        @throw [NSException exceptionWithName:NSInvalidArgumentException
+                                       reason:reason
+                                     userInfo:nil];
+    }
+
+    _maximumValue = maximumValue;
+    [self updateDisplay];
+}
+
+- (void)setStepValue:(double)stepValue {
+    if (stepValue <= 0) {
+        NSString *reason = [NSString stringWithFormat:@"Cannot set stepValue %f. Value needs to be greater than 0",
+                                    stepValue];
+        @throw [NSException exceptionWithName:NSInvalidArgumentException
+                                       reason:reason
+                                     userInfo:nil];
+    }
+
+    _stepValue = stepValue;
+    [self updateDisplay];
+}
+
+
+
+- (void)updateDisplay {
+    self.label.text = self.configuration.labelFormatter(self.value);
+    NSString *newLabel = self.configuration.accessibilityLabelFormatter(self.value);
+    self.accessibilityLabel = newLabel;
+    self.decrementButton.enabled = self.canDecrement;
+    self.incrementButton.enabled = self.canIncrement;
+
+    [self announceCurrentValue];
+}
+
+- (void)announceCurrentValue {
+    UIAccessibilityPostNotification(UIAccessibilityAnnouncementNotification, self.accessibilityLabel);
+}
+
+#pragma mark - UIAccessibilityContainer
+
+- (void)accessibilityDecrement {
+    if (self.canDecrement) {
+        [self decrementValue];
+    } else {
+        [self announceCurrentValue];
+    }
+}
+
+- (void)accessibilityIncrement {
+    if (self.canIncrement) {
+        [self incrementValue];
+    } else {
+        [self announceCurrentValue];
+    }
+}
```

Comment:
Are these used?

Reply:
In the current  implementation, yes they are.